draft-ietf-mls-protocol.md

```diff
@@ -2156,19 +2126,20 @@ A member of the group applies a Remove message by taking the following steps:
 
 ### PreSharedKey
 
-A PreSharedKey proposal can be used to request that a PreSharedKey be injected into the
-key schedule in the process of advancing the epoch.
+A PreSharedKey proposal can be used to request that a pre-shared key be
+injected into the key schedule in the process of advancing the epoch.
 
 ~~~~~
 struct {
     PreSharedKeyID psk;
 } PreSharedKey;
 ~~~~~
 
-When processing a Commit message including one or more PreSharedKey proposals, a
-group member MUST derive the `psk_secret` for the inclusion in the Key Schedule
-as described in {{pre-shared-keys}}, where the order of the PSKs corresponds to
-the order of the PreSharedKey proposals in the commit.
+The `psktype` of the pre-shared key MUST be `external` and the `psk_nonce` MUST
```

Comment:
"MUST be `external`" is too strong here; this proposal is also how "recovery" PSKs are injected, right?  I think all you want to say is "MUST NOT be `reinit`"

Reply:
There are no recovery PSKs, `external` is currently the only valid PSK type to put here

Reply:
You are right, that the of the currently specified PSK types, only `external` can be used in a psk proposal. However, implementors could define proprietary PSK types (e.g. for recovery) that they would want to use with a psk proposal as well. Not sure how we should write the spec so that option remains open.